### PR TITLE
Implemented hasprivs chat-command (Closes #7266)

### LINKED ISSUE #7266: New chat-command to list all online players with a given priv
##### Issue type

 - Feature request
 
##### Minetest version

```
0.4.16
```
##### Summary

This feature request aims to add more functionality to the `/privs` command. If the param passed with the command is, let's say `-priv`, the command accepts another param which is a priv. Now, the command should return all online players with the given privs. Here's an example for clarity:
- `/priv player` returns the privs of "player"
- `/privs -priv interact` returns all online players with the `interact` priv

##### Potential Use-cases
- Can be used by admins to check if moderators are online (if yes, then how many).
- Can be used by moderators to check which players have ____ priv.

**EDIT: This extra feature is being implemented as a separate `/hasprivs` command, which is usable only by players who are moderators or higher**

### PR DESCRIPTION
**Pull request by ClobberXD**
## Proposed Changes
- Implemented new chat-command to list all online players with a given priv.
- Minor typo fix in builtin/game/chatcommands.lua (`chattcommands -> chatcommands`).

#### Syntax of `hasprivs`
- `/hasprivs <priv>`
  - `priv` should be a registered privilege.
  - Requires `basic_privs`.

#### Sample output
- `Players (online) with the "ban" priv: Moderator1, Moderator2, Moderator3`

#### Example usage
- `/hasprivs kick`
- `/hasprivs server`

**Note: This hasn't been tested in a multi-user environment. Testing it _before_ merging is advisable, so as to verify that the code works alright in a server environment.**

### CODE REVIEW

File: builtin/game/chatcommands.lua [hunk 1]
```diff
@@ -103,6 +103,30 @@ core.register_chatcommand("privs", {
 	end,
 })
 
+core.register_chatcommand("hasprivs", {
+	params = "<priv>",
+	description = "Returns a list of all online players with the priv passed as param. e.g. /hasprivs fly",
```
Comment (Ezhh) [inline builtin/game/chatcommands.lua]: Should be "Return" not "Returns" for consistency with other descriptions. This is also a pretty long and overly technical description.
Shorter: "Return list of all online players with privilege."
If you want to include an example, the only other example in a description is in parenthesis, so it should stay consistent.

**Comment by ClobberXD** [comment]
Would it be better if any player could use this command, but they should be having the priv that they provide as param? (As was already suggested in #7266)

i.e. Only players with `fly` priv can do `/hasprivs fly`

Looks like the `/privs` command used to work the same way as described above, but was removed in this [commit](0b612539318a688ad24b66392bf39aedf8a01f62)

**Comment by Ezhh** [comment]
I'd prefer it as it is. 2 and 3 don't really make sense to me. 1 isn't bad, but I just don't see it as worth it.

**Comment by ClobberXD** [comment]
> 2 and 3 don't really make sense to me. 1 isn't bad, but I just don't see it as worth it.

2 and 3 relate to 1. They're meaningful only when 1 is implemented.

1 is just a couple of extra lines. If you like the idea, I have no issues with adding the functionality, since I feel that it's a must-have.

**Comment by Ezhh** [comment]
> I'd prefer it as it is.

**Comment by ClobberXD** [comment]
@Ezhh OK, no issues, I'll wait.

File: builtin/game/chatcommands.lua (hunk 1 above)
Comment (paramat) [reply builtin/game/chatcommands.lua]: ^ Agreed.

Comment (ClobberXD) [reply builtin/game/chatcommands.lua]: @paramat That's already been fixed in [b91f8a](https://github.com/minetest/minetest/pull/7336/commits/b91f8a4a2b922aa9bc99d5b0f89e52d45d3b196e)

File: builtin/game/chatcommands.lua [hunk 2]
```diff
@@ -103,6 +103,30 @@ core.register_chatcommand("privs", {
...
+		end
+		local privs = core.string_to_privs(param)
+		local players_with_privs = {}
+		for _, player in pairs(core.get_connected_players()) do
+			local toname = player:get_player_name()
+			if core.check_player_privs(toname, privs) then
+				table.insert(players_with_privs, toname)
+			end
+		end	
+		return true, "Players with the " ..param.. " priv: " ..table.concat(players_with_privs, ", ")
```
Comment (SmallJoker) [inline builtin/game/chatcommands.lua]: Spaces around binary operators. i.e. `"foo bar" .. "baz"`

File: builtin/game/chatcommands.lua [hunk 3]
```diff
@@ -103,6 +103,30 @@ core.register_chatcommand("privs", {
 	end,
 })
 
+core.register_chatcommand("hasprivs", {
+	params = "<priv>",
```
Comment (SmallJoker) [inline builtin/game/chatcommands.lua]: *privilege

**Comment by ClobberXD** [comment]
**New commit**: priv name in output would be surrounded by `"` for better readability. See first post for a sample output.

**Comment by ClobberXD** [comment]
#### New commits
- priv name in output would be surrounded by `"` for better readability.
- Output now makes it clear that only the online players are listed for a given priv.

See first post for a sample output.

**Pull request merged**